tower-scripts/bin/rotate-log-access-key.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+set -o errexit
+
+# A script that generates SSH keys for opsmedic's ~/.ssh/authorized_keys
+# that will be used for cluster log gathering, and rotates them keeping
+# the last key there.
+#
+# The keys managed by this script look like this in authorized_keys:
+#
+#    command="gather-logs" ssh-rsa AAA...Ob logs_access_key_week_22-YMD-H:M:S
+#
+# where '22' is the week's number. The keys are expected to be rotated weekly,
+# so this is a way to check if there's already a key for this week. If there
+# is, the default behaviour is to not rotate the keys. This check can be
+# skipped  via the --force option or by setting FORCE=true in env.
+# The 'log_access_key_week' part of the comment can be changed via the
+# --prefix= option or the KEY_COMMENT_PREFIX env var.
+# The command associated with the key can be changed with the --command=
+# option or the KEY_COMMAND env var
+#
+# The new SSH private key is printed to stdout, meant to be collected by
+# a Jenkins pipeline running this script.
+
+# TODO: this script will be typically run from a hardcoded command= in
+# use-tower's authorized_keys. We should parse SSH_ORIGINAL_COMMAND
+# for arguments
+for args in "$@"
+do
+  case $args in
+      --file=*)
+        AUTHKEYS_FILE="${args#*=}"
+        ;;
+      --command=*)
+        KEY_COMMAND="${args#*=}"
+        ;;
+      --prefix=*)
+        KEY_COMMENT_PREFIX="${args#*=}"
+        ;;
+      --force)
+        FORCE=true
+        ;;
+     --help)
+        HELP=true
+        ;;
+  esac
+done
+
+if [ "$HELP" = true ]; then
+    cat <<- EOF
+	Generate and rotate SSH keys for log gathering.
+
+	Options:
+
+	   --file=FILE      Manage FILE (default: ~opsmedic/.ssh/authorized_keys)
+	   --command=CMD    Set command=CMD in key options (default: gather-logs.sh ARG)
+	   --prefix=PREF    Use PREF_week-timestamp in the key's comment to identify it
+	   		    (default: logs_access_key_week)
+	   --force	    Do not check if there's an existing key for this week
+EOF
+    exit 0
+fi
+
+# Default values for options
+AUTHKEYS_FILE="${AUTHKEYS_FILE:-/home/opsmedic/.ssh/authorized_keys}"
+KEY_COMMENT_PREFIX="${COMMENT_PREFIX:-logs_access_key_week}"
+KEY_COMMAND="${KEY_COMMAND:-/usr/local/bin/gather-logs.sh \${SSH_ORIGINAL_COMMAND\}}"
+
+# Prepare the comment part of the ssh key to help us identify the key:
+# Keep in separate vars because "prefix_WEEKNUM" is what lets us look for the
+# presence of a key for this week
+KEY_COMMENT="${KEY_COMMENT_PREFIX}_$(date +%V)"
+KEY_TIMESTAMP=$(date +%Y%m%d-%H:%M:%S)
+
+# Exit if a key for this week is already there
+if grep -q ${KEY_COMMENT} ${AUTHKEYS_FILE} && [ -z "$FORCE" ]; then
```

Comment:
Since Jenkins will be the primary user here, would it make sense to just echo out the existing key and exit 0? That would just let us run the task again if anything happened during the push.